dbt_platform_helper/utils/git.py

```diff
@@ -6,10 +6,13 @@ def git_remote():
     git_repo = subprocess.run(
         ["git", "remote", "get-url", "origin"], capture_output=True, text=True
     ).stdout.strip()
+    return extract_repository_name(git_repo)
 
-    if not git_repo:
+
+def extract_repository_name(repository_url):
+    if not repository_url:
         return
 
-    _, repo = git_repo.split("@")[1].split(":")
+    repo = re.search(r"[^/:]*?/[^/]*?\.git\Z", repository_url).group()
```

Comment:
Maybe this could be more readable if we did it one way for repository URLs starting with "https" and the previous way for the rest?

Reply:
Only difference between the regexes would be in the [^/:] section. The rest of the regex would be the same.

Reply:
It's still not pretty, but is this easier to grok than the regex?
```suggestion
    if repository_url.startswith("https"):
        repo = repository_url.split("//")[1].split("/", maxsplit=1)[1]
    else:
        repo = repository_url.split(":")[1]
```

Reply:
I've refactored the entire method to a single line, using regex. Would be interested in thoughts on the update.

Reply:
Looking now. I was thinking we could use a capturing group to ditch that extra step in the return line, but was waiting to see whether we settled on Regex or not.